rust/processor/src/processors/parquet_processors/parquet_events_processor.rs

```diff
@@ -0,0 +1,166 @@
+    pub fn new(
+        connection_pool: ArcDbPool,
+        config: ParquetEventsProcessorConfig,
+        new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    ) -> Self {
+        if let Some(credentials) = config.google_application_credentials.clone() {
+            std::env::set_var(GOOGLE_APPLICATION_CREDENTIALS, credentials);
+        }
+
+        let event_sender = create_parquet_handler_loop::<Event>(
+            new_gap_detector_sender.clone(),
+            ProcessorName::ParquetDefaultProcessor.into(),
+            config.bucket_name.clone(),
+            config.bucket_root.clone(),
+            config.parquet_handler_response_channel_size,
+            config.max_buffer_size,
+            config.parquet_upload_interval_in_secs(),
+        );
+
+        Self {
+            connection_pool,
+            event_sender,
+        }
+    }
+}
+
+impl Debug for ParquetEventsProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ParquetProcessor {{ capacity of event channel: {:?}}}",
+            &self.event_sender.capacity(),
+        )
+    }
+}
+
+#[async_trait]
+impl ProcessorTrait for ParquetEventsProcessor {
+    fn name(&self) -> &'static str {
+        ProcessorName::ParquetEventsProcessor.into()
+    }
+
+    async fn process_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+        start_version: u64,
+        end_version: u64,
+        _: Option<u64>,
+    ) -> anyhow::Result<ProcessingResult> {
+        let last_transaction_timestamp = transactions.last().unwrap().timestamp.clone();
+        let mut transaction_version_to_struct_count: AHashMap<i64, i64> = AHashMap::new();
+
+        let mut events = vec![];
+        for txn in &transactions {
+            let txn_version = txn.version as i64;
+            let block_height = txn.block_height as i64;
+            let size_info = match txn.size_info.as_ref() {
+                Some(size_info) => size_info,
+                None => {
+                    warn!(version = txn.version, "Transaction size info not found");
+                    continue;
+                },
+            };
+            let txn_data = match txn.txn_data.as_ref() {
+                Some(data) => data,
+                None => {
+                    tracing::warn!(
+                        transaction_version = txn_version,
+                        "Transaction data doesn't exist"
+                    );
+                    PROCESSOR_UNKNOWN_TYPE_COUNT
+                        .with_label_values(&["ParquetEventsProcessor"])
+                        .inc();
+                    continue;
+                },
+            };
+            let default = vec![];
+            let raw_events = match txn_data {
+                TxnData::BlockMetadata(tx_inner) => &tx_inner.events,
+                TxnData::Genesis(tx_inner) => &tx_inner.events,
+                TxnData::User(tx_inner) => &tx_inner.events,
```

Comment:
@larry-aptos is changing this processor to handle ValidatorTransaction type. You'll probably need to pull in his changes too 

https://github.com/aptos-labs/aptos-indexer-processors/pull/454

Reply:
I will add the logic, thanks!